### PR TITLE
Add a flag to start specifying log index ranges for virtual indices.

### PR DESCRIPTION
**Pull request by dlorenc**
This is part of the temporal sharding work. The flag is not
hooked up anywhere yet.

Signed-off-by: Dan Lorenc <dlorenc@google.com>

### CODE REVIEW

File: cmd/rekor-server/app/root.go [hunk 1]
```diff
@@ -60,6 +61,8 @@ func init() {
 	rootCmd.PersistentFlags().String("trillian_log_server.address", "127.0.0.1", "Trillian log server address")
 	rootCmd.PersistentFlags().Uint16("trillian_log_server.port", 8090, "Trillian log server port")
 	rootCmd.PersistentFlags().Uint("trillian_log_server.tlog_id", 0, "Trillian tree id")
+	rootCmd.PersistentFlags().Var(&logRangeMap, "trillian_log_server.log_id_ranges", "list of tree ids and ranges")
```
Comment (bobcallaway) [inline cmd/rekor-server/app/root.go]: Is there any need to specify a different log server IP/port for each tree ID here, or is a single trillian instance sufficient for multiple trees?

File: cmd/rekor-server/app/flags_test.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	}
+}
+
+func TestLogRanges_ResolveVirtualIndex(t *testing.T) {
+	lrs := LogRanges{
+		Ranges: []LogRange{
+			{TreeId: 1, TreeLength: 17},
+			{TreeId: 2, TreeLength: 1},
+			{TreeId: 3, TreeLength: 100},
+			{TreeId: 4},
```
Comment (bobcallaway) [inline cmd/rekor-server/app/flags_test.go]: This makes sense here (for all new additions to go into `TreeId: 4`), but should we extend the command line arg to have a trailing value to match this? If the server panics and restarts, we don't want it to create a new tree but if the trillian instance reports back more than one tree that isn't specified in the list we wouldn't be able to disambiguate which one to use.

File: cmd/rekor-server/app/root.go (hunk 1 above)
Comment (bobcallaway) [inline cmd/rekor-server/app/root.go]:
```suggestion
	rootCmd.PersistentFlags().Var(&logRangeMap, "trillian_log_server.log_id_ranges", "ordered list of tree ids and ranges")
```

**Review by bobcallaway — COMMENTED** [review]
makes sense so far

File: cmd/rekor-server/app/root.go (hunk 1 above)
Comment (dlorenc) [reply cmd/rekor-server/app/root.go]: I could see us needing to eventually, but can't think of a real reason yet. WDYT? One trillian instance can handle many trees.

File: cmd/rekor-server/app/flags_test.go (hunk 2 above)
Comment (dlorenc) [reply cmd/rekor-server/app/flags_test.go]: +1, I thought about this but skipped it. We should. I'll fix it up here.

**Review by lukehinds — APPROVED** [review]
LGTM if Bob's points are resolved.

**Pull request merged**